lms/djangoapps/ccx/tests/test_models.py

```diff
@@ -125,3 +131,160 @@ def test_future_membership_disallows_auto_enroll(self):
         self.assertFalse(self.has_course_enrollment(user))
         self.assertFalse(self.has_ccx_membership(user))
         self.assertTrue(self.has_ccx_future_membership(user))
+
+
+class TestCCX(ModuleStoreTestCase):
+    """Unit tests for the CustomCourseForEdX model
+    """
+
+    def setUp(self):
+        """common setup for all tests"""
+        super(TestCCX, self).setUp()
+        self.course = course = CourseFactory.create()
+        coach = AdminFactory.create()
+        role = CourseCcxCoachRole(course.id)
+        role.add_users(coach)
+        self.ccx = CcxFactory(course_id=course.id, coach=coach)
+
+    def set_ccx_override(self, field, value):
+        override_field_for_ccx(self.ccx, self.course, field, value)
+
+    def test_ccx_course_is_correct_course(self):
+        """verify that the course property of a ccx returns the right course"""
+        expected = self.course
+        actual = self.ccx.course
+        self.assertEqual(expected, actual)
+
+    def test_ccx_course_caching(self):
+        """verify that caching the propery works to limit queries"""
+        with check_mongo_calls(1):
+            self.ccx.course
+        with check_mongo_calls(0):
+            self.ccx.course
+
+    def test_ccx_start_is_correct(self):
+        """verify that the start datetime for a ccx is correctly retrieved
+
+        Note that after setting the start field override microseconds are
+        truncated, so we can't do a direct comparison between before and after.
+        For this reason we test the difference between and make sure it is less
+        than one second.
+        """
+        expected = datetime.now(UTC())
+        self.set_ccx_override('start', expected)
+        actual = self.ccx.start
+        diff = expected - actual
+        self.assertEqual(diff.seconds, 0)
```

Comment:
Not to be pedantic, but the .days field could be non-zero.  .total_second() might be better to use.